core/src/block_data_manager/mod.rs

```diff
@@ -481,8 +481,8 @@ impl BlockDataManager {
     }
 
     pub fn block_height_by_hash(&self, hash: &H256) -> Option<u64> {
-        let result = self.block_by_hash(hash, false /* update_cache */)?;
-        Some(result.block_header.height())
+        let result = self.block_header_by_hash(hash)?;
+        Some(result.height())
```

Comment:
@peilun-conflux Do you think this might cause any issues? I think it's an overkill to read the whole block just for the height, and changing it to `block_header_by_hash` makes all related APIs usable on light nodes as well. But this call will always populate the cache.